Beiwe/Recline/Recline.swift

```diff
@@ -31,175 +29,158 @@ class Recline {
+            {
+                rObj._id = docId
+                loadedObjects.append(rObj)
             }
         }
+
         return loadedObjects
     }
 
-    // called only from _queryall, we don't need to wrap in a queue
-    func load<T: ReclineObject>(_ docId: String) -> T {
-        // give up early if db is not instantiated
-        guard let db = self.db else {
-            fatalError("what are you doing the database isn't instantiated")
-        }
-        
-        // get the _underlying_ document? it is apparently json, at least in our usage.
-        let doc: CBLDocument? = db.document(withID: docId)
-        if let doc = doc, let newMapperObj = Mapper<T>().map(JSONObject: doc.properties) {
-            // yikes wtf is this
-            newMapperObj._id = doc.properties?["_id"] as? String
-            return newMapperObj
-        } else {
-            fatalError("ok it errored. don't know what this error is though.")
-        }
-    }
-    
-    // it is annoying to get the result of a query without it all this boiler plate, and all database operations
-    // are supposed to succeed, so we wrap it and crash everything for convenience.
-    // called only from _queryall, we don't need to wrap in a queue
-    func queryOrExplode(_ view: CBLView) -> CBLQueryEnumerator {
-        let query: CBLQuery = view.createQuery()
-        var result: CBLQueryEnumerator? = nil
-        do {
-            result = try query.run()
-        } catch {
-            fatalError("your query failed? \(error)")
-        }
-        if let result = result {
-            return result
-        } else {
-            fatalError("your query completely failed? view: \(view.description), query: \(query.description)")}
-    }
-    
     // purges document from database...?
     func purge<T: ReclineObject>(_ obj: T) {
         OUTER_RECLINE_QUEUE.sync {
             return _purge(obj)
         }
     }
-    
+
     func _purge<T: ReclineObject>(_ obj: T) {
+        guard let db = self.db else {
+            fatalError("what are you doing the database isn't instantiated")
+        }
+
         if let object_id = obj._id {
             do {
-                try db?.document(withID: object_id)?.purgeDocument()
+                let doc = try db.defaultCollection().document(id: object_id)
+                try db.defaultCollection().delete(document: doc!)
             } catch {
                 fatalError("error purging document: \(error)")
-           }
+            }
         }
     }
 
     /// runs the database compact operation (why? we have a TINY database.)
     func compact() {
         OUTER_RECLINE_QUEUE.sync {
-            self._compact()
-        }
-    }
-    
-    func _compact() {
-        do {
-            try self.db?.compact()
-        } catch {
-            fatalError("error compacting database: \(error)")
+            // no available compact function on the newest version of couchbase
```

Comment:
Awesome.

Reply:
I couldn't find `compact()` function in the new version of Couchbase, thus the removal.